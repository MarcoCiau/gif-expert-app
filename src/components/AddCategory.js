import React, { useState } from "react";
import PropTypes from 'prop-types'
import "./AddCategory.css";
export const AddCategory = ({ updateCategoryHandler }) => {
  const [inputValue, setInputValue] = useState("");

  const userInputHandler = (e) => {
    setInputValue(e.target.value);
  };

  const submitFormHandler = (e) => {
    e.preventDefault();
    if (inputValue.length > 2) {
      updateCategoryHandler(inputValue);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={submitFormHandler}>
      <input type="text" value={inputValue} onChange={userInputHandler}></input>
    </form>
  );
};

AddCategory.propTypes = {
  updateCategoryHandler: PropTypes.func.isRequired,
};

AddCategory.propTypes = {
  updateCategoryHandler : PropTypes.func.isRequired
}
export default AddCategory;
