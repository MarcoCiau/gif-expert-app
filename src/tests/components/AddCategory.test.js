import React, { Component } from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import AddCategory from "../../components/AddCategory";

describe("Test with <AddCategory /> component", () => {
  const testFunc = () => {};
  const wrapper = shallow(
    <AddCategory updateCategoryHandler={testFunc}></AddCategory>
  );

  test("renders <AddCategory /> correctly ", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
