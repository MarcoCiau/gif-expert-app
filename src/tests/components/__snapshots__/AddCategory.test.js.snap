// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test with <AddCategory /> component renders <AddCategory /> correctly  1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
